oaknational/Oak-Web-Application
# fix: new search api hits with no legacy suffix - LESQ-427

## Description

Music year: 1973

- Only adds legacy prefix to legacy lesson and unit search hits 
- Adds legacy flag to each hit from fetch results 2020 and 2023
- Adds tests


## Issue(s)

Fixes #

## How to test

1. Go to https://deploy-preview-2104--oak-web-application.netlify.thenational.academy
Go to search and use flag=enable_2023_search_api 

The links should work for new lessons and units
https://deploy-preview-2104--oak-web-application.netlify.thenational.academy/teachers/search?term=Measuring+waves&flag=enable_2023_search_api

and still work for old
https://deploy-preview-2104--oak-web-application.netlify.thenational.academy/teachers/programmes/computing-secondary-ks4-l/units/programming-2-selection-cbc4/lessons


## Checklist
